### PR TITLE
manage: verbosity control

### PR DESCRIPTION
**Pull request by dachary**
## Status

Ready for review

## Description of Changes

Add logging and verbosity control (via the --verbose flag) to the manage.py subcommand. Also add pytest capture log which defines a convenient fixture for pytest to inspect logged messages.

## Testing

pytest -v pytest

## Deployment

The --verbose option is not used during deployment and has no side effects.

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

### CODE REVIEW

**Comment by dachary** [comment]
This pull request is in preparation of both [i18n](https://github.com/freedomofpress/securedrop/pull/1983) and a [manage clean_tmp bug fix](https://github.com/freedomofpress/securedrop/pull/1850). I also suspect it will be helpful in the context of Zack's work with a new manage sub-command to delete dangling submissions.

I originally made these as part of the [manage clean_tmp bug fix](https://github.com/freedomofpress/securedrop/pull/1850). But now that two pull requests depend on it, there is value in separating it to a standalone pull request. With this pull request merged we can make progress on i18n commits even if the bug fix to clean_tmp has not been resolved yet.

File: securedrop/manage.py [hunk 1]
```diff
@@ -18,8 +19,11 @@
 from db import db_session, init_db, Journalist
 from management.run import run
 
+logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s')
+log = logging.getLogger(__name__)
 
-def reset():  # pragma: no cover
+
+def reset(args):  # pragma: no cover
```
Comment (garrettr) [inline securedrop/manage.py]: Why did you add `args` to so many functions when it does not appear to be used in any of them?

File: securedrop/manage.py [hunk 2]
```diff
@@ -247,10 +256,15 @@ def get_args():
     return parser
 
 
+def setup_verbosity(args):
+    logging.getLogger(__name__).setLevel(args.verbose)
```
Comment (garrettr) [inline securedrop/manage.py]: I don't like how you're overloading `args.verbose`, which, as a flag, is implied to be a boolean, with constants indicating the log level. I think it is better to decouple "what the user asked for" from "how the user's request should be implemented internally."

File: securedrop/manage.py (hunk 1 above)
Comment (dachary) [inline securedrop/manage.py]: Because it is going to be used in both [i18n](https://github.com/freedomofpress/securedrop/pull/1983) and a [manage clean_tmp bug fix](https://github.com/freedomofpress/securedrop/pull/1850).

**Comment by dachary** [comment]
I changed the handling of --verbose to be explicit as you suggested + added a test to fix coverage. Thanks for your review :-)

File: securedrop/manage.py (hunk 1 above)
Comment (garrettr) [inline securedrop/manage.py]: Ok, but then it should be added by either of those PR's, not this one. I say this for two reasons:

1. In general, the changes in a PR should be tightly scoped and limited to only the changes necessary to address the corresponding issue. This makes reviewing, `git blame`ing, etc. much easier.
2. Adding `args` in this PR will create merge conflicts in your other PR's. I don't see why you wouldn't just wait to add it in one of the other PR's instead (where it is actually needed/used by other changes).

File: securedrop/manage.py [hunk 3]
```diff
@@ -247,10 +252,18 @@ def get_args():
     return parser
 
 
+def setup_verbosity(args):
+    if args.verbose:
+        logging.getLogger(__name__).setLevel(logging.DEBUG)
+    else:
+        logging.getLogger(__name__).setLevel(logging.INFO)
```
Comment (garrettr) [inline securedrop/manage.py]: Thanks, this is exactly what I had in mind 👍

File: securedrop/tests/test_manage.py [hunk 4]
```diff
@@ -14,6 +14,10 @@ def test_parse_args(self):
         # just test that the arg parser is stable
         manage.get_args()
 
+    def test_setup_verbose(self):
+        args = manage.get_args().parse_args(['run'])
+        manage.setup_verbosity(args)
```
Comment (garrettr) [inline securedrop/tests/test_manage.py]: How is this a test? I don't see any assertions anywhere.

Comment (dachary) [inline securedrop/tests/test_manage.py]: this is not a test, this is restoring coverage but I'll add a test as well

File: securedrop/manage.py (hunk 1 above)
Comment (dachary) [inline securedrop/manage.py]: I created this PR to reduce the scope of the other PRs and gradually merge them. I did not do that originally (this PR was part of https://github.com/freedomofpress/securedrop/pull/1850) but after a few weeks with no review activity it seemed like a good idea to split it in this way. I'm happy to re-integrate it back to https://github.com/freedomofpress/securedrop/pull/1850 if you think this is the best course of action.

Comment (dachary) [inline securedrop/manage.py]: Another reason for splitting is that the two PRs that need this change are unrelated and it would not make much sense to have one depending on the other.

**Comment by dachary** [comment]
with the tests added it looks more like a standalone pull request :-)

File: securedrop/manage.py (hunk 1 above)
Comment (garrettr) [inline securedrop/manage.py]: @dachary I'm not suggesting you re-integrate them; I think separate PR's is definitely the way to go. However, I think that you should remove irrelevant changes from this PR, such as these new `args` parameters. Does that make sense?

File: securedrop/tests/test_manage.py (hunk 4 above)
Comment (garrettr) [inline securedrop/tests/test_manage.py]: There is no such thing as restoring coverage without adding tests. It is acceptable to explicitly mark code as not covered by tests (e.g. [`# pragma: no cover`](https://coverage.readthedocs.io/en/latest/excluding.html)), if you do not think it is necessary or worthwhile to test certain sections of code; however, it is not acceptable to trick the coverage analyzer into thinking new code is covered without actually testing the new code, which is what you did here.

Comment (dachary) [inline securedrop/tests/test_manage.py]: It's better with asserts indeed. But it has value without assert. If the covered code has a typo the test will fail. Without the test a typo would go unnoticed. In any case I added the required tests.

**Comment by dachary** [comment]
I removed the commit adding arguments.

**Comment by garrettr** [comment]
In the context of #1826, this PR only affects `./manage.py` and the functional tests, so I think it is safe to merge without the CircleCI test coverage.
